machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,917 @@
+
+	// imageRegistryBundleUserData is Image Registry Data provided by the user
+	// +optional
+	ImageRegistryBundleUserData []ImageRegistryBundle `json:"imageRegistryBundleUserData"`
+
+	// imageRegistryBundleData is the ImageRegistryData
+	// +optional
+	ImageRegistryBundleData []ImageRegistryBundle `json:"imageRegistryBundleData"`
+
+	// TODO: Investigate using a ConfigMapNameReference for the PullSecret and OSImageURL
+
+	// pullSecret is the default pull secret that needs to be installed
+	// on all machines.
+	// +optional
+	PullSecret *corev1.ObjectReference `json:"pullSecret,omitempty"`
+
+	// internalRegistryPullSecret is the pull secret for the internal registry
+	// +nullable
+	InternalRegistryPullSecret []byte `json:"internalRegistryPullSecret"`
+
+	// images is map of images that are used by the controller to render templates under ./templates/
+	// +kubebuilder:validation:Required
+	Images map[string]string `json:"images"`
+
+	// BaseOSContainerImage is the new-format container image for operating system updates.
+	// +kubebuilder:validation:Required
+	BaseOSContainerImage string `json:"baseOSContainerImage"`
+
+	// BaseOSExtensionsContainerImage is the matching extensions container for the new-format container
+	// +optional
+	BaseOSExtensionsContainerImage string `json:"baseOSExtensionsContainerImage"`
+
+	// OSImageURL is the old-format container image that contains the OS update payload.
+	// +optional
+	OSImageURL string `json:"osImageURL"`
+
+	// releaseImage is the image used when installing the cluster
+	// +kubebuilder:validation:Required
+	ReleaseImage string `json:"releaseImage"`
+
+	// proxy holds the current proxy configuration for the nodes
+	// +kubebuilder:validation:Required
+	// +nullable
+	Proxy *configv1.ProxyStatus `json:"proxy"`
+
+	// infra holds the infrastructure details
+	// +kubebuilder:validation:EmbeddedResource
+	// +kubebuilder:validation:Required
+	// +nullable
+	Infra *configv1.Infrastructure `json:"infra"`
+
+	// dns holds the cluster dns details
+	// +kubebuilder:validation:EmbeddedResource
+	// +kubebuilder:validation:Required
+	// +nullable
+	DNS *configv1.DNS `json:"dns"`
+
+	// ipFamilies indicates the IP families in use by the cluster network
+	// +kubebuilder:validation:Required
+	IPFamilies IPFamiliesType `json:"ipFamilies"`
+
+	// networkType holds the type of network the cluster is using
+	// XXX: this is temporary and will be dropped as soon as possible in favor of a better support
+	// to start network related services the proper way.
+	// Nobody is also changing this once the cluster is up and running the first time, so, disallow
+	// regeneration if this changes.
+	// +optional
+	NetworkType string `json:"networkType,omitempty"`
+
+	// Network contains additional network related information
+	// +kubebuilder:validation:Required
+	// +nullable
+	Network *NetworkInfo `json:"network"`
+}
+
+// ImageRegustryBundle contains information for writing image registry certificates
```

Comment:
"ImageRegustryBundle" -> "ImageRegistryBundle" (u -> i)

Reply:
Moving [here][1]; feel free to mark this thread resolved.

[1]: https://github.com/openshift/api/pull/1453#discussion_r1337873259